sirepo/template/radia.py

```diff
@@ -418,15 +449,45 @@ def _read_or_generate(geom_id, data):
         return get_application_data(data)
 
 
-def _save_fm_sdds(name, f_data, file_path):
+def _read_solution(sim_id):
+    try:
+        return _read_path(sim_id, 'solution')
+    except KeyError:
+        # not solved yet
+        return []
+
+
+def _rot_flat_list(l, r):
+    return r.apply(numpy.reshape(l, (-1, 3)))
```

Comment:
-1 and 3 don't mean anything to me. Is -1 mean "inverse"? Does 3 mean three dimensions. Why not get those from `l`?

Reply:
-1 means "numpy will figure this dimension out automatically".  "3" does mean 3 dimensions since these are vectors.  I will rename to ```_rotate_flat_vector_list(vectors, rotation)```  which should provide context

Reply:
"rotation" is a matrix? If you call it "vectors", then "rotation" might better be "matrix", because we already know that we are rotating by the name of the function. You could also call it rotation_matrix.

Reply:
rotation is a scipy Rotation object, more general than a matrix

Reply:
then "scipy_rotation" or "numpy_rotation". Unfortunately, numpy's API doesn't define it as an object or even give it a useful name. We have to help the reader to know what the thing is.